### PR TITLE
DO-436: ShowMoreLess component added.

### PR DESCRIPTION
**Pull request by amitjain24524**

### CODE REVIEW

File: src/components/common/ShowMoreLess/ShowMoreLess.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './_ShowMoreLess.scss';
+import Container from '../Typography/Container';
+
+class ShowMoreLess extends React.Component {
```
Comment (mohitsaladiya) [inline src/components/common/ShowMoreLess/ShowMoreLess.jsx:6]: Rename it to more meaningful generic name:
```suggestion
class ExpandableText extends React.Component {
```

File: src/components/common/ShowMoreLess/_ShowMoreLess.scss [hunk 2]
```diff
@@ -0,0 +1,27 @@
+.show-more-less__description {
```
Comment (mohitsaladiya) [inline src/components/common/ShowMoreLess/_ShowMoreLess.scss:1]: Rename the classNames as per the new component name

File: src/components/common/Typography/Container/Container.jsx [hunk 3]
```diff
@@ -31,12 +32,14 @@ Container.propTypes = {
   ]),
   className: PropTypes.string,
   onClick: PropTypes.func,
+  ref: PropTypes.func,
```
Comment (mohitsaladiya) [inline src/components/common/Typography/Container/Container.jsx:35]:
```suggestion
  refFunc: PropTypes.func,

```

File: src/components/common/Typography/Container/Container.jsx [hunk 4]
```diff
@@ -31,12 +32,14 @@ Container.propTypes = {
...
   className: PropTypes.string,
   onClick: PropTypes.func,
+  ref: PropTypes.func,
 };
 
 Container.defaultProps = {
   children: null,
   className: '',
   onClick: () => {},
+  ref: () => {},
```
Comment (mohitsaladiya) [inline src/components/common/Typography/Container/Container.jsx:42]:
```suggestion
  refFunc: () => {},

```

File: src/components/common/ShowMoreLess/ShowMoreLess.jsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    return (
+      <Container>
+        <Container
+          refFunc={(ref) => { this.span = ref; }}
+          className={this.state.expanded ? 'show-more-less__description show' : 'show-more-less__description'}
+        >
+          <span>{this.props.message}</span>
+
+          <Container className={this.state.overflowActive ? 'show-more-less__more-less-button' : 'show-more-less__more-less-button hide'}>
+            <a onClick={this.showMore}>
```
[the remark this answers is not in the document]
Comment (amitjain24524) [reply src/components/common/ShowMoreLess/ShowMoreLess.jsx]: done

File: src/components/common/ShowMoreLess/ShowMoreLess.jsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      expanded: false,
+      overflowActive: false,
+    };
+    this.showMore = this.showMore.bind(this);
+  }
+
+  isEllipsisActive(e) {
```
[the remark this answers is not in the document]
Comment (amitjain24524) [reply src/components/common/ShowMoreLess/ShowMoreLess.jsx]: done

File: src/components/common/ShowMoreLess/ShowMoreLess.jsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+import Container from '../Typography/Container';
+
+class ShowMoreLess extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      expanded: false,
+      overflowActive: false,
+    };
+    this.showMore = this.showMore.bind(this);
```
[the remark this answers is not in the document]
Comment (amitjain24524) [reply src/components/common/ShowMoreLess/ShowMoreLess.jsx]: done

File: src/components/common/ShowMoreLess/ShowMoreLess.jsx [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+  isEllipsisActive(e) {
+    return e.offsetHeight < e.scrollHeight || e.offsetWidth < e.scrollWidth;
+  }
+
+  componentDidMount() {
+    this.setState({ overflowActive: this.isEllipsisActive(this.span) });
+  }
+
+  showMore() {
```
[the remark this answers is not in the document]
Comment (amitjain24524) [reply src/components/common/ShowMoreLess/ShowMoreLess.jsx]: done

File: src/components/common/ShowMoreLess/ShowMoreLess.jsx [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    ) : (
+      this.setState({ expanded: false })
+    );
+  }
+
+  render() {
+    return (
+      <Container>
+        <Container
+          refFunc={(ref) => { this.span = ref; }}
```
[the remark this answers is not in the document]
Comment (amitjain24524) [reply src/components/common/ShowMoreLess/ShowMoreLess.jsx]: done. ref remove from the container component.

File: src/components/common/ShowMoreLess/ShowMoreLess.jsx (hunk 1 above)
Comment (amitjain24524) [reply src/components/common/ShowMoreLess/ShowMoreLess.jsx]: done

File: src/components/common/ShowMoreLess/_ShowMoreLess.scss (hunk 2 above)
Comment (amitjain24524) [reply src/components/common/ShowMoreLess/_ShowMoreLess.scss]: done

File: src/components/common/Typography/Container/Container.jsx (hunk 3 above)
Comment (amitjain24524) [reply src/components/common/Typography/Container/Container.jsx]: Done ref prop remove from Container Component.

File: src/components/common/Typography/Container/Container.jsx (hunk 4 above)
Comment (amitjain24524) [reply src/components/common/Typography/Container/Container.jsx]: done

**Pull request merged**